src/dmd/backend/gloop.d

```diff
@@ -166,7 +159,15 @@ struct Iv
 nothrow:
     Symbol *IVbasic;        // symbol of basic IV
     elem **IVincr;          // pointer to parent of IV increment elem
-    famlist *IVfamily;      // variables in this family
+    Barray!famlist IVfamily;      // variables in this family
+
+    ~this()
+    {
+        foreach (ref fl; IVfamily)
+        {
+            fl.__dtor();
+        }
```

Comment:
The "fix" would be to move this to `Barray`.
```
static if (__traits(compiles, T.__dtor))
{
    foreach (ref e; array)
        e.__dtor();
}
```

Reply:
Edited to use `__xdtor` instead.  Other places where you explicitly call `__dtor` on each element before calling `array.__dtor()` can be removed as well.